Casks/visual-studio.rb

```diff
@@ -1,20 +1,27 @@
 cask "visual-studio" do
-  version "17.5.6.3,91b81b8c-74ac-4710-a93f-49bf4553845e,e5a8ce0b143da17fa397b257562e6033"
-  sha256 "296df672054773fbee2784e88b461464966bd80b3aedc84c0f5c8e77ddb2213c"
+  arch arm: "arm64", intel: "x64"
 
-  url "https://download.visualstudio.microsoft.com/download/pr/#{version.csv.second}/#{version.csv.third}/visualstudioformac-#{version.csv.first}-x64.dmg"
+  on_arm do
+    version "17.5.6.3,91b81b8c-74ac-4710-a93f-49bf4553845e,ed3a76bd92fcbfe800b830b2739774e0"
+    sha256 "cf336af2f0b93d4934ababf2d684b61da16bd8fa9a16b675291872ef26b03d7d"
+  end
+  on_intel do
+    version "17.5.6.3,91b81b8c-74ac-4710-a93f-49bf4553845e,e5a8ce0b143da17fa397b257562e6033"
+    sha256 "296df672054773fbee2784e88b461464966bd80b3aedc84c0f5c8e77ddb2213c"
+  end
+
+  url "https://download.visualstudio.microsoft.com/download/pr/#{version.csv.second}/#{version.csv.third}/visualstudioformac-#{version.csv.first}-#{arch}.dmg"
   name "Microsoft Visual Studio"
   desc "Integrated development environment"
   homepage "https://visualstudio.microsoft.com/vs/mac/"
 
   livecheck do
-    url "https://aka.ms/vs/mac/download"
-    regex(%r{/download/pr/([^/]+)/([^/]+)/visualstudioformacinstaller[._-]v?(\d+(?:\.\d+)+).dmg}i)
-    strategy :header_match do |headers, regex|
-      match = headers["location"].match(regex)
-      next if match.blank?
-
-      "#{match[3]},#{match[1]},#{match[2]}"
+    url "https://aka.ms/vsmac/manifest/17-stable"
```

Comment:
@razvanazamfirei Should we use the below, would require a manual bump for major version

```suggestion
    url "https://aka.ms/vsmac/manifest/#{version.major}-stable"
```

Reply:
@bevanjkay, sure! I hope it's still going to work with a new version, but worth a shot!